### PR TITLE
Only query genes that are oncokb curated

### PR DESCRIPTION
**Pull request by adamabeshouse**

### CODE REVIEW

**Comment by adamabeshouse** [comment]
@onursumer please pay attention especially to how Im handling things in AnnotationColumnFormatters.getData, make sure I'm not messing anything up there.

File: src/pages/patientView/copyNumberAlterations/column/AnnotationColumnFormatter.tsx [hunk 1]
```diff
@@ -27,15 +28,22 @@ export default class AnnotationColumnFormatter
 
         if (copyNumberData)
         {
-            let oncoKbIndicator: IndicatorQueryResp|undefined;
-
-            if (oncoKbData && oncoKbData.result && oncoKbData.status === "complete") {
-                oncoKbIndicator = AnnotationColumnFormatter.getIndicatorData(copyNumberData, oncoKbData.result);
+            let oncoKbIndicator: IndicatorQueryResp|undefined = undefined;
+            let oncoKbStatus:"complete"|"pending"|"error" = "complete";
+            const oncoKbGeneExist = !!oncoKbAnnotatedGenes[copyNumberData[0].entrezGeneId];
```
Comment (alisman) [inline src/pages/patientView/copyNumberAlterations/column/AnnotationColumnFormatter.tsx]: i think you could use "in" operator.  slightly more semantic.  minor quibble

File: src/pages/patientView/copyNumberAlterations/column/AnnotationColumnFormatter.tsx [hunk 2]
```diff
@@ -27,15 +28,22 @@ export default class AnnotationColumnFormatter
 
         if (copyNumberData)
         {
-            let oncoKbIndicator: IndicatorQueryResp|undefined;
-
-            if (oncoKbData && oncoKbData.result && oncoKbData.status === "complete") {
-                oncoKbIndicator = AnnotationColumnFormatter.getIndicatorData(copyNumberData, oncoKbData.result);
+            let oncoKbIndicator: IndicatorQueryResp|undefined = undefined;
+            let oncoKbStatus:"complete"|"pending"|"error" = "complete";
```
Comment (alisman) [inline src/pages/patientView/copyNumberAlterations/column/AnnotationColumnFormatter.tsx]: we've got to have this in once place somewhere.  we use it so much.

Comment (adamabeshouse) [reply src/pages/patientView/copyNumberAlterations/column/AnnotationColumnFormatter.tsx]: youre right

File: src/pages/patientView/copyNumberAlterations/column/AnnotationColumnFormatter.tsx (hunk 1 above)
Comment (adamabeshouse) [reply src/pages/patientView/copyNumberAlterations/column/AnnotationColumnFormatter.tsx]: do we have a good Set object?

File: src/shared/components/annotation/OncoKB.tsx [hunk 3]
```diff
@@ -24,6 +24,7 @@ export interface IOncoKbProps {
     evidenceCache?: OncoKbEvidenceCache;
     evidenceQuery?: Query;
     pubMedCache?: OncokbPubMedCache;
+    geneNotExist:boolean;
```
Comment (alisman) [inline src/shared/components/annotation/OncoKB.tsx]: geneNotAnnotated?

Comment (adamabeshouse) [reply src/shared/components/annotation/OncoKB.tsx]: geneExist is the keyword used in the oncokb API, so I used exist for coherence

File: src/pages/patientView/clinicalInformation/PatientViewPageStore.ts [hunk 4]
```diff
@@ -474,15 +475,22 @@ export class PatientViewPageStore {
...
         await: () => [
+            this.oncoKbAnnotatedGenes,
             this.mutationData,
             this.uncalledMutationData,
             this.clinicalDataForSamples,
             this.studiesForSamplesWithoutCancerTypeClinicalData,
             this.studies
         ],
-        invoke: async() => fetchOncoKbData(this.sampleIdToTumorType, this.mutationData, this.uncalledMutationData),
+        invoke: () => {
```
[the remark this answers is not in the document]
Comment (adamabeshouse) [reply src/pages/patientView/clinicalInformation/PatientViewPageStore.ts]: there is no await done inside so theres no need

**Pull request merged**